app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryViewModel.kt

```diff
@@ -37,11 +40,15 @@ class PromotedStoryViewModel(
   }
 
   fun clickOnStoryTile() {
-    routeToTopicPlayStoryListener.routeToTopicPlayStory(
-      internalProfileId,
-      promotedStory.topicId,
-      promotedStory.storyId
-    )
+    if (myDownloadsFeatureFlagValue)
+      routeToTopicPreviewListener.routeToTopicPreview(internalProfileId, promotedStory.topicId)
```

Comment:
Ditto here (please make sure there are tests covering this & its opposite cases).